src/plugins/midi-piano/midi-piano-display.js

```diff
@@ -0,0 +1,292 @@
+    const noteName = getNoteNameFromMidiValue(message.data[1]);
+    const velocity = message.data.length > 2 ? message.data[2] : 0;
+    const command = message.data[0];
+    const eventType = getEventTypefromMidiCommand(command, velocity);
+    const midiValue = message.data[1];
+    updateKeyVisualization(eventType, midiValue);
+    handleSamplerEvent(eventType, noteName);
+  }
+
+  function onMIDISuccess(midiAccess) {
+    midiAccessObj.current = midiAccess;
+    for (const input of midiAccessObj.current.inputs.values()) {
+      input.onmidimessage = handleMidiDeviceEvent;
+    }
+  }
+
+  function onMIDIFailure() {
+    // eslint-disable-next-line no-console
+    console.log('Could not access your MIDI devices.');
+  }
+
+  function handleMidiPlayerEvent(message) {
+
+    const velocity = message.velocity;
+    if (message.name !== 'Note on' && message.name !== 'Note off') {
+      return;
+    }
+    const eventType = velocity > 0 ? 'Note on' : 'Note off';
+    const noteName = message.noteName;
+    const midiValue = message.noteNumber;
+    updateKeyVisualization(eventType, midiValue);
+    handleSamplerEvent(eventType, noteName);
+  }
+
+  function instantiatePlayer() {
+    if (player.current) {
+      return;
+    }
+    player.current = new midiPlayerNs.Player();
+    player.current.on('midiEvent', message => {
+      handleMidiPlayerEvent(message);
+    });
+    player.current.on('playing', () => {
+
+    });
+    player.current.on('endOfFile', () => {
+      player.current.stop();
+    });
+    player.current.loadArrayBuffer(midiData.current);
+  }
+
+  const removeActiveClassFromKeys = () => {
+    for (const key of keys.current) {
+      if (key && key.classList.contains('ReactPiano__Key--active')) {
+        key.classList.remove('ReactPiano__Key--active');
+      }
+    }
+  };
+
+  const startMidiPlayer = () => {
+
+    if (player.current !== null && player.current.isPlaying()) {
+      return;
+    }
+    if (player.current === null) {
+      instantiatePlayer();
+      player.current.play();
+      return;
+    }
+    player.current.play();
+  };
+
+  const pauseMidiPlayer = () => {
+    if (!player.current.isPlaying()) {
+      return;
+    }
+    player.current.pause();
+    removeActiveClassFromKeys();
+    sampler.current.releaseAll();
+  };
+
+  const stopMidiPlayer = () => {
+    player.current.stop();
+    removeActiveClassFromKeys();
+    sampler.current.releaseAll();
+  };
+
+  const playNote = midiValue => {
+    if (!samplerHasLoaded.current) {
+      return;
+    }
+    sampler.current.triggerAttack(getNoteNameFromMidiValue(midiValue));
+  };
+
+  const stopNote = midiValue => {
+    setTimeout(() => {
+      sampler.current.triggerRelease(getNoteNameFromMidiValue(midiValue));
+    }, 200);
+  };
+
+  const getData = () => {
+    try {
+      axios.get(src, { responseType: 'arraybuffer' })
+        .then(response => {
+          midiData.current = response.data;
+        });
```

Comment:
As we have the `HttpClient` wrapping the `axios` usages, let's please use the `HttpClient` for this request and not `axios` directly.
The reason for this is that, if needed, we could easily replace `axios` with a different module with very little code change.